int32 time
string type
string desc
int16 robotID
